pennylane/transforms/zx/converter.py

```diff
@@ -0,0 +1,587 @@
+# Copyright 2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Transforms for interacting with PyZX, framework for ZX calculus."""
+# pylint: disable=too-many-statements, too-many-branches, too-many-return-statements, too-many-arguments
+
+from collections import OrderedDict
+import numpy as np
+import pennylane as qml
+from pennylane.transforms.op_transforms import op_transform
+from pennylane.tape import QuantumScript
+from pennylane.wires import Wires
+
+
+class VertexType:  # pylint: disable=too-few-public-methods
+    """Type of a vertex in the graph.
+
+    This class is copied from PyZX as we do not make PyZX a Pennylane requirement.
+
+    Copyright (C) 2018 - Aleks Kissinger and John van de Wetering"""
+
+    BOUNDARY = 0
+    Z = 1
+    X = 2
+    H_BOX = 3
+
+
+class EdgeType:  # pylint: disable=too-few-public-methods
+    """Type of an edge in the graph.
+
+    This class is copied from PyZX as we do not make PyZX a Pennylane requirement.
+
+    Copyright (C) 2018 - Aleks Kissinger and John van de Wetering"""
+
+    SIMPLE = 1
+    HADAMARD = 2
+
+
+@op_transform
+def to_zx(qscript, expand_measurement=False):  # pylint: disable=unused-argument
+    """This transform converts a PennyLane quantum script to a ZX-Graph in the `PyZX framework <https://pyzx.readthedocs.io/en/latest/>`_.
+    The graph can be optimized and transformed by well-known ZX-calculus reductions.
+
+    Args:
+        qscript(QuantumScript, Operation): The PennyLane quantum circuit.
+        expand_measurements(bool): The expansion will be applied on measurements that are not in the Z-basis and
+            rotations will be added to the operations.
+
+    **Example**
+
+    You can use the transform decorator directly on your ``QNode``, and executing it will produce a PyZX graph.
```

Comment:
```suggestion
    You can use the transform decorator directly on your ``QNode``, quantum function, or :class:`~.QuantumScript`, and executing it will produce a PyZX graph.
```

Reply:
I think you can use the transform on the QuantumScript but not the decorator right? I will rewrite it.

Reply:
```suggestion
    You can use the transform decorator directly on your ``QNode`` or quantum function, and executing it will produce a PyZX graph. You can also use the transform directly on a :class:`~.QuantumScript`.
```